### ISSUE TITLE
Reproducing Results

### ISSUE THREAD

**Issue by kmo6rqb4** [cdb342/GDFusion#3]
Hi,

Thank you again for the great work!

I'm trying to reproduce the results in your paper. I run:
```bash
bash tools/dist_train.sh configs/gdfusion/alocc_3d_r50_256x704_bevdet_preatrain_gdfusion.py work_dirs/alocc3d_gdfusion 4
```

And I got these results, which are lower than ones achieved using the pretrained model.
```
===> per class IoU of 6019 samples:
===> others - IoU = 14.1688
===> barrier - IoU = 49.8234
===> bicycle - IoU = 29.7008
===> bus - IoU = 44.235
===> car - IoU = 54.8861
===> construction_vehicle - IoU = 29.0534
===> motorcycle - IoU = 31.339
===> pedestrian - IoU = 30.5016
===> traffic_cone - IoU = 37.4556
===> trailer - IoU = 33.0104
===> truck - IoU = 41.7388
===> driveable_surface - IoU = 83.9175
===> other_flat - IoU = 39.834
===> sidewalk - IoU = 56.1092
===> terrain - IoU = 59.2028
===> manmade - IoU = 52.0839
===> vegetation - IoU = 46.1143
===> free - IoU = 92.3503
===> mIoU of 6019 samples: 43.13
===> occupied - IoU = 75.7633
===> mIoU_D = 36.8082
```

Any idea why the results are lower?

**Comment by kmo6rqb4** [comment]
> Since the training is iteration-based, using 4 GPUs with a batch size of 2 per GPU means the model is trained on half the total number of samples compared to our default setting (8 GPUs x batch size 2). Please try doubling your batch size to 4 per GPU if using 4 GPUs. Moreover, the repository was recently refactored to eliminate a significant amount of redundant code, and while it reproduces the original inference results, potential issues could still be present. Please let me know if your problem continues.

I also updated the total number of iteration based on the batch size, so it should be consistent with using 8 GPUs. Maybe the learning rate should be updated as well when using less GPUs?

I'll try with 8 GPUs just to be sure.

**Comment by cdb342** [comment]
> Short update:
> 
> I was evaluating the "latest.pth" model instead of the "iter_xxxxx_ema.pth". Evaluating using the EMA model, the performance are higher.
> 
> However the performance using 4 GPUs was still lower that the one I achieved when training with 8 GPUs:
> 
> With 4 GPUs:
> ```
> ===> per class IoU of 6019 samples:
> ===> others - IoU = 15.3586
> ===> barrier - IoU = 52.3978
> ===> bicycle - IoU = 33.0959
> ===> bus - IoU = 46.0747
> ===> car - IoU = 57.0132
> ===> construction_vehicle - IoU = 31.0744
> ===> motorcycle - IoU = 33.6842
> ===> pedestrian - IoU = 33.9276
> ===> traffic_cone - IoU = 38.0786
> ===> trailer - IoU = 38.1161
> ===> truck - IoU = 44.1926
> ===> driveable_surface - IoU = 84.7502
> ===> other_flat - IoU = 49.702
> ===> sidewalk - IoU = 58.9569
> ===> terrain - IoU = 62.4311
> ===> manmade - IoU = 55.4843
> ===> vegetation - IoU = 49.7579
> ===> free - IoU = 92.5742
> ===> mIoU of 6019 samples: 46.12
> ===> occupied - IoU = 77.1466
> ===> mIoU_D = 39.6474
> ```
> 
> With 8 GPUs:

Can you share your config files?

**Comment by kmo6rqb4** [comment]
I'm using exactly the one provided in the repo without any change except changing `num_gpus` for the experiment with 4 GPUs.

**Comment by kmo6rqb4** [comment]
> > I'm using exactly the one provided in the repo without any change except changing `num_gpus` for the experiment with 4 GPUs.
> 
> As I mentioned before, the total batch size (num_gpus * batch_size_per_gpu) needs to stay the same to get consistent results. So, when using 4 GPUs, you should set the batch size to 4.

Why tho? The number of iterations depend on both number of GPUs and samples per GPU, so the total number of iterations should be consistent.
```
num_iters_per_epoch = int(28130 // (num_gpus * samples_per_gpu) * 4.554)
```
